### PR TITLE
CoreOS updates to v0.11.0 and kubelet args fix

### LINKED ISSUE #4614: Pod containers with no environment variables set
Possibly related to #4415 

Since 0.10.1 - and just confirmed with 0.11.0 - containers from pods do not have any Kubernetes related env vars set.
```
$ vagrant ssh node-01
CoreOS alpha (593.0.0)
Update Strategy: No Reboots
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                        COMMAND                CREATED             STATUS              PORTS               NAMES
0522aeebbdda        pires/elasticsearch:master   "/usr/bin/runsvdir -   47 minutes ago      Up 47 minutes                           k8s_elasticsearch-master.6d831f7e_elasticsearch-master-fplln.default.etcd_f40cf7be-b848-11e4-895a-080027aeee49_a251d366
823d55b2770c        kubernetes/pause:go          "/pause"               58 minutes ago      Up 58 minutes                           k8s_POD.8f3eed67_elasticsearch-master-fplln.default.etcd_f40cf7be-b848-11e4-895a-080027aeee49_017ece96
core@node-01 ~ $ docker exec 0522aeebbdda env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=elasticsearch-master-fplln
HOME=/root
JAVA_HOME=/usr/lib/jvm/java-8-oracle
ES_PKG_NAME=elasticsearch-1.4.2
```

@sub-mod is facing the same issue and we both can confirm that 0.9.x works well.

### PR DESCRIPTION
**Pull request by pires**
Upgraded Kubernetes to 0.11.0.
Replaced kubelet argument --etcd_servers with --api_servers.
Fixes #4614

### CODE REVIEW

File: docs/getting-started-guides/aws/cloudformation-template.json [hunk 1]
```diff
@@ -379,13 +379,13 @@
...
           "        EnvironmentFile=/etc/network-environment\n",
-          "        ExecStartPre=/usr/bin/wget -N -P /opt/bin https://storage.googleapis.com/kubernetes-release/release/v0.10.1/bin/linux/amd64/kubelet\n",
+          "        ExecStartPre=/usr/bin/wget -N -P /opt/bin https://storage.googleapis.com/kubernetes-release/release/v0.11.0/bin/linux/amd64/kubelet\n",
           "        ExecStartPre=/usr/bin/chmod +x /opt/bin/kubelet\n",
           "        ExecStart=/opt/bin/kubelet \\\n",
           "        --address=0.0.0.0 \\\n",
           "        --port=10250 \\\n",
           "        --hostname_override=${DEFAULT_IPV4} \\\n",
-          "        --etcd_servers=http://", {"Fn::GetAtt" :["KubernetesMasterInstance" , "PrivateIp"]}, ":4001\\\n",
+          "        --api_servers=", {"Fn::GetAtt" :["KubernetesMasterInstance" , "PrivateIp"]}, ":8080\\\n",
```
Comment (erictune) [inline docs/getting-started-guides/aws/cloudformation-template.json]: This should be port 7080.
Port 8080 has no authentication.
Port 8080 should not be reachable from minions.

File: docs/getting-started-guides/coreos/cloud-configs/node.yaml [hunk 2]
```diff
@@ -92,13 +92,13 @@ coreos:
...
         EnvironmentFile=/etc/network-environment
-        ExecStartPre=/usr/bin/wget -N -P /opt/bin https://storage.googleapis.com/kubernetes-release/release/v0.10.1/bin/linux/amd64/kubelet
+        ExecStartPre=/usr/bin/wget -N -P /opt/bin https://storage.googleapis.com/kubernetes-release/release/v0.11.0/bin/linux/amd64/kubelet
         ExecStartPre=/usr/bin/chmod +x /opt/bin/kubelet
         ExecStart=/opt/bin/kubelet \
         --address=0.0.0.0 \
         --port=10250 \
         --hostname_override=$private_ipv4 \
-        --etcd_servers=http://<master-private-ip>:4001 \
+        --api_servers=<master-private-ip>:8080 \
```
Comment (erictune) [inline docs/getting-started-guides/coreos/cloud-configs/node.yaml]: Same comment.

File: docs/getting-started-guides/aws/cloudformation-template.json (hunk 1 above)
Comment (erictune) [inline docs/getting-started-guides/aws/cloudformation-template.json]: I'm willing to merge this, but I want to understand why this is working.
On GCE, apiserver only listed on localhost:8080, last I checked.
I'm not sure why this is working.

Comment (pires) [inline docs/getting-started-guides/aws/cloudformation-template.json]: > On GCE, apiserver only listed on localhost:8080, last I checked.

So it makes sense to configure ```kubelet``` to reach 8080, right?

**Comment by pires** [comment]
@erictune it surely works this way.

File: docs/getting-started-guides/aws/cloudformation-template.json (hunk 1 above)
Comment (erictune) [inline docs/getting-started-guides/aws/cloudformation-template.json]: I don't think a kubelet on node x can reach apiserver which is serving on localhost:8080 of node y.

**Comment by erictune** [comment]
Oh.  This is working because this sets `--address=0.0.0.0".
Other distros are setting `--address=127.0.0.1`.
This is a mess, but now I understand why this works and it fixes stuff so I'll merge it.
Thanks for the fix.

**Comment by pires** [comment]
@erictune I understand things are not perfect with current CoreOS set-up, but it does work and is up-to-date sooner than other community-maintained Kubernetes guides. I am open to suggestions and test PRs from others, ofc.

**Comment by erictune** [comment]
@pires Your work on on CoreOS is much appreciated.  My comment about *this is a mess* was directed at myself and the mess I created when I introduced multiple ways to do master-to-node communication/auth.

**Comment by pires** [comment]
@erictune thank you for your kind words. But please, do share your ideas and let's improve this.

**Pull request merged**